Allow specifying the amount of memory for lambda function

With only 128MB of memory allocated to the lambda, some invocations are failing.  This seems to be a problem for accounts that generate large amounts of events, as the error is being thrown on the [line where the gzipped log files are being read](https://github.com/cloudandthings/terraform-aws-clickops-notifier/blob/main/lambda/app/main.py#L121). (Well, when it tries to parse the _contents_ of the log files as JSON, but you know what I mean)

```text
[ERROR] MemoryError
Traceback (most recent call last):
  File "/var/task/main.py", line 146, in handler
    raise e
  File "/var/task/main.py", line 121, in handler
    event_json = json.load(fh)
  File "/var/lang/lib/python3.8/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/var/lang/lib/python3.8/json/__init__.py", line 343, in loads
    s = s.decode(detect_encoding(s), 'surrogatepass')
```

--- a/README.md
+++ b/README.md
@@ -96,6 +96,7 @@ resource "aws_s3_bucket" "test_bucket" {
 | <a name="input_iam_role_arn"></a> [iam\_role\_arn](#input\_iam\_role\_arn) | Existing IAM role ARN for the lambda. Required if `create_iam_role` is set to `false` | `string` | `""` | no |
 | <a name="input_included_accounts"></a> [included\_accounts](#input\_included\_accounts) | List of accounts that be scanned to manual actions. If empty will scan all accounts. | `list(string)` | `[]` | no |
 | <a name="input_included_users"></a> [included\_users](#input\_included\_users) | List of emails that be scanned to manual actions. If empty will scan all emails. | `list(string)` | `[]` | no |
+| <a name="input_lambda_memory_size"></a> [lambda\_memory\_size](#input\_lambda\_memory\_size) | The amount of memory for Lambda to use | `number` | `"128"` | no |
 | <a name="input_lambda_runtime"></a> [lambda\_runtime](#input\_lambda\_runtime) | The lambda runtime to use | `string` | `"python3.8"` | no |
 | <a name="input_log_retention_in_days"></a> [log\_retention\_in\_days](#input\_log\_retention\_in\_days) | Number of days to keep CloudWatch logs | `number` | `14` | no |
 | <a name="input_message_format"></a> [message\_format](#input\_message\_format) | Where do you want to send this message? slack or msteams | `string` | `"slack"` | no |

--- a/README.md
+++ b/README.md
@@ -121,7 +121,7 @@ resource "aws_s3_bucket" "test_bucket" {
 
 | Name | Version |
 |------|---------|
-| <a name="provider_aws"></a> [aws](#provider\_aws) | 4.37.0 |
+| <a name="provider_aws"></a> [aws](#provider\_aws) | 4.38.0 |
 ----
 ### Requirements
 

--- a/README.md
+++ b/README.md
@@ -188,6 +188,8 @@ locals {
 
     "kms.amazonaws.com:Decrypt",
     "kms.amazonaws.com:RetireGrant",
+
+    "trustedadvisor.amazonaws.com:RefreshCheck",
   ]
 }
 ```

--- a/excluded_scoped_actions.tf
+++ b/excluded_scoped_actions.tf
@@ -43,5 +43,7 @@ locals {
 
     "kms.amazonaws.com:Decrypt",
     "kms.amazonaws.com:RetireGrant",
+    
+    "trustedadvisor.amazonaws.com:RefreshCheck",
   ]
 }

--- a/excluded_scoped_actions.tf
+++ b/excluded_scoped_actions.tf
@@ -43,7 +43,7 @@ locals {
 
     "kms.amazonaws.com:Decrypt",
     "kms.amazonaws.com:RetireGrant",
-    
+
     "trustedadvisor.amazonaws.com:RefreshCheck",
   ]
 }

--- a/main.tf
+++ b/main.tf
@@ -11,7 +11,7 @@ module "clickops_notifier_lambda" {
   source_path = "${path.module}/lambda/app"
 
   timeout     = var.event_processing_timeout
-  memory_size = 128
+  memory_size = var.lambda_memory_size
 
   create_role = var.create_iam_role
   lambda_role = var.iam_role_arn

--- a/variables.tf
+++ b/variables.tf
@@ -129,6 +129,12 @@ variable "lambda_runtime" {
   }
 }
 
+variable "lambda_memory_size" {
+  type        = number
+  description = "The amount of memory for Lambda to use"
+  default     = "128"
+}
+
 # IAM Setup
 variable "create_iam_role" {
   description = "Determines whether a an IAM role is created or to use an existing IAM role"

